### PR TITLE
system_console: add link to mavlink shell in QGC

### LINKED ISSUE #557: Hint to QGC console is missing in console guide
@bkueng Could you do a quick pass?

https://dev.px4.io/en/debug/system_console.html

### PR DESCRIPTION
**Pull request by bkueng**
Fixes  #557

### CODE REVIEW

File: en/debug/system_console.md [hunk 1]
```diff
@@ -11,9 +11,7 @@ There are multiple shells, but only one console: The system console is the locat
 
 > **Info**
 > USB shell: To just run a few quick commands or test an application connecting to the USB
-> shell is sufficient. The Mavlink shell can be used for this, see below.
-> The hardware serial console is only needed for boot debugging or when USB should be used
-> for MAVLink to connect a [GCS](../qgc/README.md).
+> shell is sufficient. The Mavlink shell can be used for this, see [below](#mavlink_shell).
```
Comment (hamishwillee) [inline en/debug/system_console.md]: MAVLink

Comment (bkueng) [reply en/debug/system_console.md]: Done

Comment (hamishwillee) [reply en/debug/system_console.md]: It predates your change, but this tip, and indeed the whole section here (System Console vs. Shell) is unclear.

For the tip,it says "The Mavlink shell can be used for this", but "this" is not defined. I think you are saying that the MAVLink shell can also be used for quick tests?

For the system console vs shell thing I think this is saying that a console is a type of output that is accessed through a shell. You can get this from several different shells - USB, MAVlink whatever. It is also called the NSH console. Is that correct?

If so, would bullet point all the consoles, including MAVLink, then have info at the end explaining what they are good for.

Comment (bkueng) [reply en/debug/system_console.md]: > I think you are saying that the MAVLink shell can also be used for quick tests?

Correct.

There are a few differences between the system console and the shell.
System console:
- runs on one specific UART (debug port, as configured in NuttX), used for deeper debugging/devel (bootup, NuttX, startup scripts, board bringup, development on central parts of PX4, such as uORB).
- commonly attached via FTDI cable
- you see the boot output (and all the auto-started tasks)

Whereas the shell is more high-level and can also be used through MAVLink (when there was no mavlink shell yet, it was used directly via USB for example). You'll see the output of what you started, but not of what's already running (but you won't see the output of a task running on the work queue).

Do you want to make a suggestion, so it's understandable for you?

Comment (hamishwillee) [reply en/debug/system_console.md]: I'll commit a change for you to check.

Comment (hamishwillee) [reply en/debug/system_console.md]: Done!

File: en/debug/system_console.md [hunk 2]
```diff
@@ -2,18 +2,21 @@
...
 
-  * System console (first shell): Hardware serial port
-  * Additional shells: Pixhawk on USB (e.g. lists as /dev/tty.usbmodem1 on Mac OS)
+Shells provide higher-level access to the system:
+- Used for basic module testing/running commands.
+- Only display the output of modules you start (and therefore cannot debug the boot process).
+- Cannot display the output of tasks running on the work queue.
+
+There are several shells:
+* USB shell (e.g. Pixhawk on USB lists as `/dev/tty.usbmodem1` on Mac OS)
```
Comment (bkueng) [inline en/debug/system_console.md]: This also runs mavlink now. We can write that the shell can be used on any UART (including USB), but running mavlink is more powerful and thus we only use the shell via mavlink today.

**Review by bkueng — COMMENTED** [review]
Thanks @hamishwillee

File: en/debug/system_console.md (hunk 2 above)
Comment (hamishwillee) [reply en/debug/system_console.md]: Can you update appropriately? What is wrong with this (as it is now) is that is isn't "really" clear how you would use it - ie I know how to connect up to use with QGC, but how do I use the USB option?

Comment (bkueng) [reply en/debug/system_console.md]: Done

**Comment by hamishwillee** [comment]
Thanks @bkueng ! Merged.

**Pull request merged**